packages/components/src/internal/models.ts

```diff
@@ -419,6 +420,16 @@ export class EditorModel
                     } else {
                         row = row.set(col.name, values.size === 1 ? values.first().raw : undefined);
                     }
+                } else if (col.jsonType === 'date' && !displayValues) {
+                    let dateVal;
+                    if (values.size === 1) {
+                        dateVal = values.first().raw;
+                        dateVal = parseDate(dateVal);
+                        if (!dateVal) {
+                            dateVal = parseDate(values.first().raw, getDateTimeFormat());
```

Comment:
why not always passing in getDateTimeFormat()? How would parseDate parse an ambiguous string like 01/01/2020?

Reply:
Since the input field in the editable grid is just a text input, users can supply dates in any format they like (for better or for worse), and moment apparently tries a number of formats when attempting to parse dates.  If we supply a single format, it does a worse job in this flexible scenario.  For example, 
```
> parseDate("14:07 2020-02-03")
Mon Feb 03 2020 14:07:00 GMT-0800 (Pacific Standard Time)

> parseDate("14:07 2020-02-03", "HH:mm YYYY-MM-DD")
Mon Feb 03 2020 14:07:00 GMT-0800 (Pacific Standard Time)

> parseDate("2020-02-03 14:07")
Mon Feb 03 2020 14:07:00 GMT-0800 (Pacific Standard Time)

> parseDate("2020-02-03 14:07", "HH:mm YYYY-MM-DD")
Thu Dec 24 2020 20:07:00 GMT-0800 (Pacific Standard Time)
```
I think the best answer for disambiguating when inputting dates is to provide an unambiguous string.  So, for the infamous Month-Day vs. Day-Month conundrum, you can use the name of the month:
```
> parseDate("2020-02-March 14:07")
Mon Mar 02 2020 14:07:00 GMT-0800 (Pacific Standard Time)

> parseDate("2020-Feb-03 14:07")
Mon Feb 03 2020 14:07:00 GMT-0800 (Pacific Standard Time)
```
Using the single date format string that's meant for displaying the date will never do the right thing in that case:
```
> parseDate("2020-Feb-03 14:07", "HH:mm YYYY-MM-DD")
Thu Dec 24 2020 20:07:00 GMT-0800 (Pacific Standard Time)
```

It seems, in fact, unlikely that this `if` block will produce a valid date when the call to `parseDate` without the format does not, but I leave it in for now.

Moment's documentation on [parsing](https://momentjs.com/guides/#/parsing/strict-mode/) indicates that "In a later release, the parser will default to using strict mode.", in which case we'll likely want to consider changing the editable grid so the input field for dates is not just a text box. But for now, using the `getDateTimeFormat` sparingly when parsing dates seems the most robust way to go.